Services/DeviceMethods.cs

```diff
@@ -79,21 +79,27 @@ public class DeviceMethods : IDeviceMethods
 
         public Task<MethodResponse> ExecuteMethodAsync(MethodRequest methodRequest, object userContext)
         {
-            this.log.Info("Creating task to execute method with json payload.", () => new
+            try
             {
-                this.deviceId,
-                methodName = methodRequest.Name,
-                methodRequest.DataAsJson
-            });
+                this.log.Info("Creating task to execute method with json payload.", () => new
+                {
+                    this.deviceId,
+                    methodName = methodRequest.Name,
+                    methodRequest.DataAsJson
+                });
+
+                // Kick the method off on a separate thread & immediately return
+                // Not immediately returning would block the client connection to the hub
+                var t = Task.Run(() => this.MethodExecution(methodRequest));
 
-            // Kick the method off on a separate thread & immediately return
-            // Not immediately returning would block the client connection to the hub
-            var t = Task.Run(() => this.MethodExecution(methodRequest));
+                return Task.FromResult(new MethodResponse((int) HttpStatusCode.OK));
```

Comment:
so we won't return a message like earlier "Executed Method: ...". Was that an issue?